tests/ui/impl-trait/in-trait/object-safety.rs

```diff
@@ -19,4 +19,5 @@ fn main() {
     //~| ERROR the trait `Foo` cannot be made into an object
     let s = i.baz();
     //~^ ERROR the trait `Foo` cannot be made into an object
+    //~| ERROR the trait bound `dyn Foo: Foo`
```

Comment:
This "regression" is due to the fact that the RPITIT is now a GAT (since it captures lifetimes now). GATs hit this code:

https://github.com/rust-lang/rust/blob/f525bb4e2afac6d2491a86c178aa98e3f3931130/compiler/rustc_trait_selection/src/traits/select/confirmation.rs#L547-L553